api/container/containerv1/subnets.go

```diff
@@ -25,10 +25,18 @@ type SubnetProperties struct {
 	Gateway           string `json:"gateway"`
 }
 
+type UserSubnet struct {
+	CIDR   string `json:"cidr" binding:"required" description:"The CIDR of the subnet that will be bound to the cluster. Eg.format: 12.34.56.78/90"`
+	VLANID string `json:"vlan_id" binding:"required" description:"The private VLAN where the CIDR exists'"`
+}
+
 //Subnets interface
 type Subnets interface {
 	AddSubnet(clusterName string, subnetID string, target ClusterTargetHeader) error
 	List(target ClusterTargetHeader) ([]Subnet, error)
+	AddClusterUserSubnet(clusterID string, userSubnet UserSubnet, target ClusterTargetHeader) error
+	ListClusterUserSubnets(clusterID string, target ClusterTargetHeader) ([]Vlan, error)
```

Comment:
Is it supposed to return subnets or Vlans?

Reply:
https://containers.bluemix.net/swagger-api/#!/clusters/GetClusterUserSubnet shows response type as `string` which is definitely not right. Issue has been created to rectify that.